ui/apps/routing/src/extensions/breadcrumb-navigation.tsx

```diff
@@ -0,0 +1,97 @@
+import { FormNext } from 'grommet-icons';
+import singleSpaReact from 'single-spa-react';
+import * as React from 'react';
+import ReactDOM from 'react-dom';
+import { RootExtensionProps } from '@akashaorg/typings/ui';
+import DS from '@akashaorg/design-system';
+import { withProviders, ThemeWrapper } from '@akashaorg/ui-awf-hooks';
+import { I18nextProvider } from 'react-i18next';
+import * as singleSpa from 'single-spa';
+
+const { Icon, BasicCardBox, Box, Text, styled } = DS;
+
+const StyledLink = styled(Text)`
+  color: ${({ theme }) => theme.colors.grey};
+  cursor: pointer;
+
+  &:hover {
+    color: ${({ theme }) => theme.colors.accentText};
+  }
+`;
+
+const matchUrlToRoute = (url: string, routes: Record<string, string>) => {
+  const route = Object.keys(routes).find(route => url.includes(routes[route]));
+  return route;
+};
+
+const stackLimit = 4;
+const BreadcrumbNavigation: React.FC<RootExtensionProps> = props => {
+  const [stack, setStack] = React.useState([location.href]);
+
+  React.useEffect(() => {
+    window.addEventListener('single-spa:before-routing-event', (evt: CustomEvent) => {
```

Comment:
This listener supports also setting a signal, you can use `AbortController` to call `.abort()` when you need to remove the listener. This is a better approach so you don't end up cleaning another listener.